compiler/rustc_ast/src/ast.rs

```diff
@@ -1795,6 +1795,26 @@ pub enum LitIntType {
     Unsuffixed,
 }
 
+impl LitIntType {
+    pub const fn max_literal_value(&self) -> u128 {
+        match self {
+            LitIntType::Signed(IntTy::I8) => i8::MAX as u128,
+            LitIntType::Signed(IntTy::I16) => i16::MAX as u128,
+            LitIntType::Signed(IntTy::I32) => i32::MAX as u128,
+            LitIntType::Signed(IntTy::I64) => i64::MAX as u128,
+            LitIntType::Signed(IntTy::I128) => i128::MAX as u128,
+            LitIntType::Signed(IntTy::Isize) => isize::MAX as u128,
```

Comment:
This uses the host's `isize`, but should be using use the target's `isize`. Same for `usize` below.

Reply:
Fair. What should I do to obtain these values for the target?

Reply:
You can get size of `isize` and `usize` from `tcx.data_layout.pointer_size`. That will return a [`Size`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_abi/struct.Size.html), which has the [`signed_int_max`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_abi/struct.Size.html#method.signed_int_max) and [`unsigned_int_max`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_abi/struct.Size.html#method.unsigned_int_max) methods that seem useful. It looks like the nearest `tcx` is the [one in the `LoweringContext`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast_lowering/struct.LoweringContext.html#structfield.tcx), which you as `self` from here: https://github.com/rust-lang/rust/blob/b0ae21860621da0f2efd6c08dc1f8af7ab2fee55/compiler/rustc_ast_lowering/src/format.rs#L15